api/usermanager/user_manager.go

```diff
@@ -0,0 +1,204 @@
+package usermanager
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/bixlabs/authentication/api/usermanager/structures/create"
+	"github.com/bixlabs/authentication/api/usermanager/structures/delete"
+	findOne "github.com/bixlabs/authentication/api/usermanager/structures/findone"
+	"github.com/bixlabs/authentication/api/usermanager/structures/mappers"
+	"github.com/bixlabs/authentication/api/usermanager/structures/update"
+	"github.com/bixlabs/authentication/authenticator/interactors"
+	"github.com/bixlabs/authentication/authenticator/interactors/implementation/util"
+	"github.com/gin-gonic/gin"
+)
+
+type userManagerRESTConfigurator struct {
+	userManager interactors.UserManager
+}
+
+func NewUserManagerRESTConfigurator(userManager interactors.UserManager, engine *gin.Engine) {
+	configureUserManagerRoutes(userManagerRESTConfigurator{userManager}, engine)
+}
+
+func configureUserManagerRoutes(restConfig userManagerRESTConfigurator, r *gin.Engine) *gin.Engine {
+	router := r.Group("/v1/users")
+
+	router.POST("/search", restConfig.findOne)
```

Comment:
`/search` sounds like a big name for this, shouldn't be something like `/find_by_email`? (I don't know the convention that we are using for naming endpoints, consider that).

Another thing, I don't think this should be a POST, but a GET, are you doing this to prevent showing the email in the URL?

Reply:
Good point! I think the same /search is  abig name for a simple endpoint

Yes, First  tried to use GET but swagger denies all GET request with a body, they don't allow those calls. But also the email is sensitive information, therefore it shouldn't be visible on the URL